client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -0,0 +1,501 @@
+	SUBMITTING_PAYMENT_KEY_REQUEST,
+	SUBMITTING_WPCOM_REQUEST,
+} from 'lib/store-transactions/step-types';
+
+const debug = debugFactory( 'calypso:checkout:payment:apple-pay' );
+
+/**
+ * Web Payment method identifier.
+ */
+export const WEB_PAYMENT_BASIC_CARD_METHOD = 'basic-card';
+export const WEB_PAYMENT_APPLE_PAY_METHOD = 'https://apple.com/apple-pay';
+
+/**
+ * Supported card types.
+ */
+const SUPPORTED_NETWORKS = [ 'visa', 'mastercard', 'discover', 'amex', 'jcb', 'maestro' ];
+const APPLE_PAY_MERCHANT_IDENTIFIER = config( 'apple_pay_merchant_id' );
+const PAYMENT_REQUEST_OPTIONS = {
+	requestPayerName: true,
+	requestPayerPhone: false,
+	requestPayerEmail: false,
+	requestShipping: false,
+};
+
+/**
+ * Returns an available Web Payment method.
+ *
+ * Web Payments (`PaymentRequest` API) are available only if the
+ * document is served through HTTPS.
+ *
+ * The configuration features `my-sites/checkout/web-payment/*` must
+ * also be enabled.
+ *
+ * @returns {string|null}  One of the `WEB_PAYMENT_*_METHOD` or `null`
+ *                         if none can be detected.
+ */
+export function detectWebPaymentMethod() {
+	if ( ! config.isEnabled( 'my-sites/checkout/web-payment' ) ) {
+		return null;
+	}
+
+	if (
+		config.isEnabled( 'my-sites/checkout/web-payment/apple-pay' ) &&
+		window.ApplePaySession &&
+		window.ApplePaySession.canMakePayments()
+	) {
+		return WEB_PAYMENT_APPLE_PAY_METHOD;
+	}
+
+	if ( config.isEnabled( 'my-sites/checkout/web-payment/basic-card' ) && window.PaymentRequest ) {
+		return WEB_PAYMENT_BASIC_CARD_METHOD;
+	}
+
+	return null;
+}
+
+/**
+ * Returns a user-friendly Web Payment method name.
+ *
+ * @param {string|null} webPaymentMethod  The payment method identifier
+ *                                        (expecting one of the
+ *                                        `WEB_PAYMENT_*_METHOD`
+ *                                        constant).
+ * @returns {string|null}                 A user-friendly payment name
+ *                                        or the given payment method
+ *                                        if none matches.
+ */
+export function getWebPaymentMethodName( webPaymentMethod ) {
+	switch ( webPaymentMethod ) {
+		case WEB_PAYMENT_BASIC_CARD_METHOD:
+			return i18n.translate( 'Browser wallet' );
+
+		case WEB_PAYMENT_APPLE_PAY_METHOD:
+			return 'Apple Pay';
+
+		default:
+			return webPaymentMethod;
+	}
+}
+
+/**
+ * The `<WebPaymentBox />` component.
+ */
+export class WebPaymentBox extends React.Component {
+	static propTypes = {
+		cart: PropTypes.object.isRequired,
+		transaction: PropTypes.object.isRequired,
+		transactionStep: PropTypes.object.isRequired,
+		countriesList: PropTypes.array.isRequired,
+		onSubmit: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	state = {
+		country: null,
+		postalCode: null,
+	};
+
+	/**
+	 * @return {object} A dictionnary containing `default`, `disabled` and `text` keys.
+	 */
+	buttonState = () => {
```

Comment:
I think this function should be called `getButtonState()` so it's easier to differentiate (like during a search) from the variable that holds its result.

Reply:
Done, thanks!